packages/react-client/src/hooks/useConnection.ts

```diff
@@ -1,41 +1,64 @@
-import type { GenericMetadata } from "@fishjam-cloud/ts-client";
+import type { GenericMetadata, ReconnectionStatus } from "@fishjam-cloud/ts-client";
 import { useCallback } from "react";
 
+import type { PeerStatus } from "../types/public";
 import { useFishjamContext } from "./internal/useFishjamContext";
 import { useReconnection } from "./internal/useReconnection";
 
 export interface JoinRoomConfig<PeerMetadata extends GenericMetadata = GenericMetadata> {
   /**
-   * fishjam URL
+   * Fishjam URL
    */
   url: string;
   /**
-   * token received from server (or Room Manager)
+   * Token received from server (or Room Manager)
    */
   peerToken: string;
   /**
-   * string indexed record with metadata, that will be available to all other peers
+   * String indexed record with metadata, that will be available to all other peers
    */
   peerMetadata?: PeerMetadata;
 }
 
+/**
+ * @category Connection
+ */
+export interface UseConnectionResult {
```

Comment:
Introducing such type is not helpful in terms of code documentation. You can compare mobile/web documentation.

Mobile still have 'old' syntax and it looks like this: https://docs.fishjam.io/api/mobile/functions/useConnection.
And for Web, with this updated syntax it looks like this: https://docs.fishjam.io/api/web/functions/useConnection